File: OfflineClient/app/src/main/java/de/jbamberger/offlineclient/di/DatabaseModule.java
package de.jbamberger.offlineclient.di;

import android.arch.persistence.room.Room;
import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import de.jbamberger.offlineclient.db.OfflineDatabase;
import de.jbamberger.offlineclient.db.dao.RedditPostDao;

/**
 * Dependency Injection module to provide database related objects.
 *
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
public class DatabaseModule {

    private static final String DB_NAME = "offline-fetcher-app-db";

    @Provides
    @Singleton
    OfflineDatabase providesDb(Context context) {
        return Room.databaseBuilder(context, OfflineDatabase.class, "main.db").build();
    }

    @Provides
    @Singleton
    RedditPostDao providesRedditPostDao(OfflineDatabase database) {
        return database.redditPostDao();
    }
}

File: OfflineClient/cached-api/src/main/java/de/jbamberger/api/db/DatabaseModule.kt
package de.jbamberger.api.db

import android.app.Application
import android.arch.persistence.room.Room
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Dependency Injection module to provide database related objects.
 *
 * @author Jannik Bamberger (dev.jbamberger@gmail.com)
 */

@Module
class DatabaseModule {

    @Provides
    @Singleton
    internal fun providesDb(context: Application): OfflineDatabase {
        return Room.databaseBuilder(context, OfflineDatabase::class.java, DB_NAME).build()
    }

    companion object {
        private const val DB_NAME = "main.sqlite"
    }
}
